frontend/src/App.jsx

```diff
@@ -42,22 +42,22 @@ function App() {
                 </ProtectedRoute>
               } />
               <Route path="/add-trips" element={
-                <ProtectedRoute>
```

Comment:
Sorry I may not had explained myself well enough. Those routes should still be protected, but I just think we should avoid re-fetching data from ``/users/auth-check`` when we switch section.

For example:

If a user opens the `/add-trips ` route initially, we should perform the request to  ``/users/auth-check``. If user is authenticated, we store their info in context.

If they navigate to another section (e.g. /past-trips) we just check if user is authenticated from context, without fetching data from ``/users/auth-check` (unless they refresh the page). Does that make sense? :)